.github/workflows/doc.yml

```diff
@@ -0,0 +1,53 @@
+on:
+  push:
+    branches:
+      - master
+      - '0.[0-9]+.x'
+      - '1.[0-9]+.x'
+
+name: Publish Docs
+jobs:
+  publish_docs:
+    if: github.repository == 'diesel-rs/diesel'
+    name: Publish Docs
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v2
+    - name: Cache cargo registry
+      uses: actions/cache@v2
+      with:
+        path: ~/.cargo/registry
+        key: cargo-doc-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Cache cargo index
+      uses: actions/cache@v2
+      with:
+        path: ~/.cargo/git
+        key: cargo-doc-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        profile: minimal
+        override: true
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install libmysqlclient-dev libsqlite3-dev libpq-dev
+    - name: Build documentation
+      uses: actions-rs/cargo@v1
+      with:
+        command: doc
+        args: --manifest-path diesel/Cargo.toml --features "postgres sqlite mysql extras"
```

Comment:
As we host the docs ourself I think we want to include dependencies here, because otherwise for example the docs for the derive macros will not show up correctly.

Reply:
Then I think `--workspace` is a better way, how about?